### ISSUE TITLE
Manage state to persist repo values across pages

### ISSUE THREAD

**Issue by MichaelCurrin** [MichaelCurrin/badge-generator#117]
Use a store pattern.

The store could also get and set in the local storage, as a wrapper.

https://michaelcurrin.github.io/code-cookbook/recipes/javascript/packages/vue/state-management/store-pattern.html

Ideas of fields to store:

- Repo name. Useful across pages at least.
- Repo username. Useful across pages and across site visits so it is personalized.

**Comment by JerryHue** [comment]
I would like to follow up with some questions I have.

So, I implemented the store to track the username and repo name when they are submitted for the first time. Thus, if you create a repo badge, the username and the repo name will persist and will show up as well for the package badge form. I suppose this is how the feature works.

Another thing I am concerned with is whether this "global" won't be shared across other users. I assume that this value is local to the session, so my changes will not collide with other users' submissions. I went ahead and opened the same website on two different browser windows (one incognito and one normal), and it seemed that the values were not overwriting each other. However, I am not sure why this is the case. I am not sure if we are using sessions or something like that.

Let me know if you have any comments

**Comment by MichaelCurrin** [comment]
## Independence 

 The current site resets the forms whenever you switch pages which is annoying.

So yes the first benefit I want from the store is that you can reuse repo fields on multiple pages/views.

No there is no danger of different users sharing the same values. There is no backend or database to persist data. So all the logic and store happens on the browser and for that user independent of others.

## Tabs and stores

In the simple store case (no local storage or session storage, the store is just a JavaScript object on the page.

An over simplification:

```js
const store = {}
```

If open another _tab_ where incognito or not,  the browser will load the entire index.html and vue app _and_ set up a new `store` object. So if you have 5 tabs open then you have 5 stores. If you close and open the browser, the store will be lost and start from empty because the JS object is lost.

This is all fine behavior for now.

## Advanced store

Once the basic store logic works, then a part two in a separate PR can use localStorage which is like a cookie but has no expiry date and holds more data

The MDN (Mozilla) docs cover localStorage well.

The advantages of localStorage here:

- if you enter form values, they will persist across views as before
- you will have _one_ global store across different tabs on the same window. I don't know how incognito works here but it doesn't matter as this is not sensitive financial data or similar.
- when you close and open the tab or browser window, the store will persist.

You should also know about sessionStorage which works identically to localStorage except it will _clear_ when you close the browser. Same as session cookies. For this app, localStorage makes more sense.

**Comment by JerryHue** [comment]
Thank you for your throughout response! I will follow up with my PR shortly, thank you for your help and guidance.

**Comment by MichaelCurrin** [comment]
I've merged the PR.

I moved the store.ts module up a level and renamed the variables and methods to be consistent with the rest of the app.

I also added tests.


See #126

**Comment by MichaelCurrin** [comment]
I just created PR #131 - @JerryHue I would appreciate your review or input, maybe try it locally.

So now local storage is used!



<img width="914" alt="Screen Shot 2021-11-20 at 5 07 24 pm" src="https://user-images.githubusercontent.com/18750745/142731355-482a48ce-7ab7-45b1-8596-13cf4ed5cd98.png">


BTW I dropped use of `state` and `reactive` as they gave incorrect behavior when used with `localStorage`.
